[FEAT] Drop support for Tracy

ktf opened the pull request:
[FEAT] Drop support for Tracy

alibuild commented:
Error while checking build/O2/fullCI for 46d28a27e2472cf3e7215e7c467b2aa7d3cffbfb at 2024-03-05 21:39:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2-full-system-test-latest/log
task timeout reached .. killing all processes


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/83ca26d425cfe0064d4ad4d67ddc283f267423d1/slc8_x86-64/o2checkcode/1.0-local173/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12745/46d28a27e2472cf3e7215e7c467b2aa7d3cffbfb/build_O2_fullCI/pretty.html).

ktf committed:
DPL: drop tracy support for good

ktf committed:
DPL: do not use Tracy anymore

Tracy usage never took off, so I am removing all the explicit calls to it and
I will replace them with Signpost usage, where it still makes sense.

ktf committed:
DPL: drop tracy support from the GUI

No one is using it and Signposts are probably a better alternative.

ktf committed:
[FEAT] Drop support for Tracy

ktf committed:
DPL: make sure O2_SIGNPOST_ID_FROM_POINTER allows for const pointers

ktf committed:
DPL: rename TracyLockable to O2_LOCKABLE

ktf committed:
DPL: drop unused Tracy APIs

ZoneScoped*, TracyAppInfo, FrameMark, TracyAlloc, TracyPlot were all removed
and will be replaced by Signposts as needed.

ktf committed:
DPL: use signposts for OOB callback

ktf committed:
DPL: replace Tracy allocation profiler with Signposts
